server/src/main/java/org/elasticsearch/plugins/IngestPlugin.java

```diff
@@ -27,4 +30,8 @@ public interface IngestPlugin {
     default Map<String, Processor.Factory> getProcessors(Processor.Parameters parameters) {
         return Map.of();
     }
+
+    default Optional<Pipeline> getIngestPipeline(IndexMetadata indexMetadata, Processor.Parameters parameters) {
+        return Optional.empty();
+    }
```

Comment:
What happens if multiple plugins define an ingest pipeline? Should it be disallowed?